### PR TITLE
fix: pasting after select all or part of a table, meaningless tables and rows are created (close #378)

### LINKED ISSUE #378: In wysiwyg, if pasting after select all or part of a table, meaningless tables and rows are created.
## Version

after this commit https://github.com/nhnent/tui.editor/commit/df974bd815a9f14cc002b97c412b3298a7ab2e7d

## Test Environment

Chrome

## Current Behavior

screen recording 2019-01-29 at 11 44 45 am mov

## Expected Behavior

Case 1. Table is selected all with other lines.
image To Be => image

Case 2. Part of table is selected with other lines.
image To Be => image

image To Be => image

### PR DESCRIPTION
**Pull request by sohee-lee7**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has a description of the breaking change

### Description

#378 

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/js/wwClipboardManager.js [hunk 1]
```diff
@@ -102,11 +102,21 @@ class WwClipboardManager {
     const range = this.wwe.getEditor().getSelection();
     const tableManager = this.wwe.componentManager.getManager('table');
 
-    if (tableManager.isInTable(range)) {
+    if (tableManager.isInTable(range) && this._isInOneCell(range)) {
       this._tablePasteHelper.pasteClipboard(ev);
     }
   }
 
+  _isInOneCell(range) {
```
Comment (junghwan-park) [inline src/js/wwClipboardManager.js]: `_isSingleCellSelected()`는 어떤가요?

Comment (sohee-lee7) [reply src/js/wwClipboardManager.js]: 좀 더 명시적이네요 ㅎㅎ 변경하겠습니다!

File: src/js/wwTableManager.js [hunk 2]
```diff
@@ -832,14 +837,13 @@ class WwTableManager {
...
         return;
-      } else if (node.nodeName === 'TABLE'
-                && $node.find('thead').length === 0
-                && $node.find('tbody').length === 0
-      ) {
-        $node.remove();
       }
 
-      this._completeIncompleteTable(node);
+      if (node.nodeName === 'TABLE' && $node.find('tbody').length === 0) {
```
[the remark this answers is not in the document]
Comment (sohee-lee7) [reply src/js/wwTableManager.js]: thbody의 경우는 복구해야하는 경우입니다.

image
위와 같이 선택해서 backspce 시에 스콰이어에 의해서 테이블 헤더가 사라지고 바디만 남게 됩니다.
이때는 이전과 같이 복구 루틴을 타게 됩니다.

File: src/js/wwTableManager.js [hunk 3]
```diff
@@ -709,6 +709,11 @@ class WwTableManager {
     this._addTrIntoContainerIfNeed($table);
 
     const $trs = $table.find('tr');
+    $trs.each((index, tr) => {
+      if (!tr.cells.length) {
+        tr.remove();
+      }
+    });
```
[the remark this answers is not in the document]
Comment (sohee-lee7) [reply src/js/wwTableManager.js]: 좋은 방법입니다!! 반영하겠습니다 +_+

**Pull request merged**